packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/athena/start-query-execution.ts

```diff
@@ -66,10 +76,21 @@ export class AthenaStartQueryExecution extends sfn.TaskStateBase {
     this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
 
     validatePatternSupported(this.integrationPattern, AthenaStartQueryExecution.SUPPORTED_INTEGRATION_PATTERNS);
+    this.validateExecutionParameters(props.executionParameters);
 
     this.taskPolicies = this.createPolicyStatements();
   }
 
+  private validateExecutionParameters(executionParameters?: string[]) {
+    if (executionParameters == null) return;
```

Comment:
```suggestion
    if (executionParameters === undefined || Token.isUnresolved(executionParameters)) return;
```
Validation should be skipped also when the parameter is a Token.
For example when specified as [`JsonPath.listAt()`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_stepfunctions.JsonPath.html#static-listwbratpath).

Reply:
I have a question about that actually... I think right now this will fail if using things like `JsonPath.listAt()`, because the typescript schema says `string[]` and that will not match. I don't know enough of the machinery to say what it should be – something like `string[] | Token`?

Reply:
Nah that's fine. Tokens are pretending to be strings and string arrays.

Reply:
No, actually `JsonPath.listAt()` returns `string[]`. But `JsonPath.array()` returns `string`. Not sure if there's anything to fix on that front or not.

Reply:
`executionParameters` is `string[]` so using `JsonPath.array()` will not produce an accepted value in this case. 
It's up to the user to specify the correct `JsonPath` function, we don't need to support every `JsonPath` returned type.